arch/cpu/cc2538/rtimer-arch.c

```diff
@@ -70,21 +70,17 @@ rtimer_arch_init(void)
 void
 rtimer_arch_schedule(rtimer_clock_t t)
 {
-  rtimer_clock_t now;
-
   /* STLOAD must be 1 */
   while((REG(SMWDTHROSC_STLOAD) & SMWDTHROSC_STLOAD_STLOAD) != 1);
 
   INTERRUPTS_DISABLE();
 
-  now = RTIMER_NOW();
-
   /*
    * New value must be 5 ticks in the future. The ST may tick once while we're
    * writing the registers. We play it safe here and we add a bit of leeway
    */
-  if((int32_t)(t - now) < 7) {
-    t = now + 7;
+  if(!rtimer_is_schedulable(t)) {
+    t = RTIMER_NOW() + RTIMER_GUARD_TIME;
```

Comment:
There is something that feels unnatural to me here. If we abstract away the notion of schedulability, why do we still compute `t` manually here? What if `rtimer_is_schedulable` gets modified in the future with criteria different from just looking at the guard? So, if we are to add this `rtimer_is_schedulable`, maybe it also makes sense to also add a function that returns the next scheduleable time.

Reply:
My motivation to expose `rtimer_is_schedulable` was that my current CSL implementation depends on it. Yet, I found a workaround for this function ... so I removed this function from this PR